spec/lib/mdot/client_spec.rb

```diff
@@ -112,6 +112,58 @@
         end
       end
     end
+
+    context 'temporary/permanent address' do
+      before do
+        VCR.insert_cassette(
+          cassette,
+          match_requests_on: %i[method uri headers],
+          erb: { icn: user.icn }
+        )
+      end
+
+      after { VCR.eject_cassette }
+
+      context 'with a null temporary_address value from system of record' do
+        let!(:cassette) { 'mdot/get_supplies_null_temp_address_200' }
+
+        it 'returns a valid response' do
+          response = subject.get_supplies
+          expect(response).to be_ok
+          expect(response).to be_an MDOT::Response
+        end
+      end
+
+      context 'with a no temporary_address from system of record' do
+        let(:cassette) { 'mdot/get_supplies_no_temp_address_200' }
+
+        it 'returns a valid response' do
+          response = subject.get_supplies
+          expect(response).to be_ok
+          expect(response).to be_an MDOT::Response
+        end
+      end
+
+      context 'with a null permanent_address value from system of record' do
+        let(:cassette) { 'mdot/get_supplies_null_perm_address_200' }
+
+        it 'returns a valid response' do
+          response = subject.get_supplies
+          expect(response).to be_ok
+          expect(response).to be_an MDOT::Response
+        end
+      end
+
+      context 'with a no permanent_address from system of record' do
+        let(:cassette) { 'mdot/get_supplies_no_perm_address_200' }
+
+        it 'returns a valid response' do
+          response = subject.get_supplies
+          expect(response).to be_ok
+          expect(response).to be_an MDOT::Response
+        end
+      end
```

Comment:
Note that the "no `{permanent,temporary}_address` test scenarios may not be valid. Remove once confirmed.